### PR TITLE
Add property to allow turn on KMS-SSE encryption when putting data in S3

### PR DESCRIPTION
**Pull request by haroldb1**
Add property to allow turn on KMS-SSE encryption when putting data in S3 with the given kms id.  Update README.md

### CODE REVIEW

**Comment by haroldb1** [comment]
Hi @rhusar please could you help review this.

Thanks,
Harold

**Comment by rhusar** [comment]
Thanks @haroldpirum – I did review the code, I think this generally looks good, just need to run the tests to verify there isn't anything going awry. Just some nitpicks. Il l test this ASAP.

File: src/main/java/org/jgroups/protocols/aws/S3_PING.java [hunk 1]
```diff
@@ -68,8 +69,13 @@ public class S3_PING extends FILE_PING {
         "on each update. This is useful in multi-region deployments where each region exists in its own AWS account.")
     protected boolean acl_grant_bucket_owner_full_control = false;
 
+    @Property(description="Use kms encryption with s3 with the given kms key (optionally - enables KMS Server side encryption (SSE-KMS) using the given kms key", exposeAsManagedAttribute=false)
```
Comment (rhusar) [inline src/main/java/org/jgroups/protocols/aws/S3_PING.java]: This is missing closing `)`.

File: src/main/java/org/jgroups/protocols/aws/S3_PING.java [hunk 2]
```diff
@@ -68,8 +69,13 @@ public class S3_PING extends FILE_PING {
...
+    protected String  kms_key_id;
+
     protected AmazonS3 s3;
 
+    protected SSEAwsKeyManagementParams encryptionParams = null;
```
Comment (rhusar) [inline src/main/java/org/jgroups/protocols/aws/S3_PING.java]: The initialization of a field is redundant as per JLS.

**Review by rhusar — CHANGES_REQUESTED** [review]

**Comment by haroldb1** [comment]
hi @rhusar could you help review again please.
Thanks!

**Comment by haroldb1** [comment]
hi @rhusar i believe i added all the requested changes, though the review still says changes requested.  Is it possible to recheck please.

Thanks,
Harold

**Comment by rhusar** [comment]
@haroldpirum Sorry for the delay, we needed to clear up a few things before really opening the branch for further development. Will review and merge this week.

**Comment by haroldb1** [comment]
Hi @rhusar are we in a position to merge this please?

**Comment by ioannoapirum** [comment]
@rhusar any chance this could be looked at? It would benefit our project.

**Comment by rhusar** [comment]
@haroldpirum @ioannoapirum Sorry for the delay I will try to review by tomorrow.

**Review by rhusar — APPROVED** [review]
The changes look good now. Just to mention - anyone looking at this feature - this is a *server-side* encryption. Also, Amazon S3 applies server-side encryption with Amazon S3 managed keys (SSE-S3) as the base level of encryption for every bucket in Amazon S3 since start of 2023.

**Comment by rhusar** [comment]
Thanks @haroldpirum!

**Comment by rhusar** [comment]
Oh, would a release be helpful @haroldpirum @ioannoapirum ?

**Pull request merged**